From c6e38e23c1a2d45efcab73793ebe28f8fcb504e4 Mon Sep 17 00:00:00 2001
From: "M. Vefa Bicakci" <vefa.bicakci@windriver.com>
Date: Fri, 4 Feb 2022 19:44:06 -0500
Subject: [PATCH] bnxt_en/bnxt_compat.h: Fix up a build failure

This commit resolves the following build failure:

  In file included from ./include/net/sock.h:59,
                   from ./include/linux/tcp.h:19,
                   from ./include/linux/ipv6.h:87,
                   from ./include/net/dsfield.h:12,
                   from ./include/net/ip_tunnels.h:13,
                   from ./include/net/dst_metadata.h:6,
                   from /builddir/build/BUILD/netxtreme-bnxt_en-1.10.2-219.0.55.0/bnxt_en-1.10.2-219.0.55.0/bnxt.h:30,
                   from /builddir/build/BUILD/netxtreme-bnxt_en-1.10.2-219.0.55.0/bnxt_en-1.10.2-219.0.55.0/bnxt_hwrm.c:26:
  ./include/linux/filter.h:980:1: error: useless type name in empty declaration [-Werror]
   void bpf_warn_invalid_xdp_action(u32 act);
   ^~~~
  cc1: all warnings being treated as errors

The problem arises from the fact that the function
bpf_warn_invalid_xdp_action is already defined in the kernel. Having a
placeholder macro definition with the same name causes the compilation
warning (which turns into an error due to -Werror).

Signed-off-by: M. Vefa Bicakci <vefa.bicakci@windriver.com>
---
 bnxt_en-1.10.2-220.0.13.0/bnxt_compat.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/bnxt_en-1.10.2-220.0.13.0/bnxt_compat.h b/bnxt_en-1.10.2-220.0.13.0/bnxt_compat.h
index 63d27f2e8979..aeb15fc9ab2a 100644
--- a/bnxt_en-1.10.2-220.0.13.0/bnxt_compat.h
+++ b/bnxt_en-1.10.2-220.0.13.0/bnxt_compat.h
@@ -1596,7 +1596,6 @@ enum xdp_action {
 
 #ifndef HAVE_BPF_TRACE
 #define trace_xdp_exception(dev, xdp_prog, act)
-#define bpf_warn_invalid_xdp_action(act)
 #endif
 
 #ifdef HAVE_XDP_RXQ_INFO
-- 
2.29.2

